### DISCUSSION TITLE
Threshold BLS Signatures and DKG in native IOS app

### DISCUSSION THREAD

**Discussion by sreedeeppaul** [celo-org/social-connect#234] [Help (Q&A)]
Hello has someone been able to reference static library( libblind_threshold_bls.a ) inside their IOS Swift app from the wrappers written here https://github.com/valora-inc/react-native-blind-threshold-bls

**Comment by arthurgousset** [comment]
Hey @sreedeeppaul, thanks for posting here!

Could you share any errors you are encountering when referencing the library? That might help us understand the issue better 😊 
Thanks!

**Reply by sreedeeppaul** [reply]
Hey @0xarthurxyz  I want to know if anyone has been able to use the Rust library in native Swift ios app for creating Threshold BLS Signatures used in ODIS Salt generation
https://github.com/celo-org/celo-threshold-bls-rs
We have some react native wrappers here https://github.com/valora-inc/react-native-blind-threshold-bls for IOS but I am looking for a native Swift wrapper without React component consuming the Rust libraray
Wanted to understand from the community if anyone was able to directly use the Rust library in their native code

**Reply by arthurgousset** [reply]
Thanks for asking! Unfortunately I don't know 🤷‍♂️ 
Perhaps folks that came across this in the future can chime in.
